app/api/groups/get_participant_progress.go

```diff
@@ -91,73 +91,84 @@ type rawParticipantProgressRaw struct {
 	IsParent bool
 }
 
+type participantProgressParameters struct {
+	ItemID                     int64
+	ParticipantID              int64
+	CheckPermissionsForGroupID int64
+	ParticipantType            string
+	GetChildren                bool
+}
+
 // swagger:operation GET /items/{item_id}/participant-progress groups groupParticipantProgress
 //
-//		---
-//		summary: Get progress of a participant
-//		description: >
-//	             Returns the current progress of a participant on a given item.
+//	---
+//	summary: Get progress of a participant
+//	description: >
+//		Returns the current progress of a participant on a given item.
+//
 //
+//		For `{item_id}` and all its visible children,	displays the results of the given participant
+//		(current user or `as_team_id` (if given) or `watched_group_id` (if given)).
 //
-//	             For `{item_id}` and all its visible children,
-//	             displays the results of the given participant
-//	             (current user or `as_team_id` (if given) or `watched_group_id` (if given)).
-//	             Only one of `as_team_id` and `watched_group_id` can be given.
-//	             The results are sorted by `items_items.child_order`.
+//		The children are returned only if the current user has a started result on the given `item_id`.
```

Comment:
Have you changed the doc to reflect this? I would expect no "children" key in the output when not allowed. (to be done if it does not complicate things too much / if it does not add a new sql request)

Reply:
fixed, now the "children" key appears only if the current user have a started result on the item_id, tested with `And the response should not be defined at $.children`. There's also a test with "children": [], when there is no children but the current user has a started result.